packages/typespec-go/src/lib.ts

```diff
@@ -117,7 +117,32 @@ const EmitterOptionsSchema: JSONSchemaType<GoEmitterOptions> = {
 
 const libDef = {
   name: '@azure-tools/typespec-go',
-  diagnostics: {},
+  diagnostics: {
+    'InternalError': {
+      severity: 'error',
+      messages: {
+        default: paramMessage`The emitter encountered an internal error during preprocessing. Please open an issue at https://github.com/Azure/autorest.go/issues and include the complete error message.\n${'stack'}`
```

Comment:
Should we differentiate between an internal error and a `not implemented` error? I'm not convinced one way or the other so this is just something to think about.